File: FinPlanner/FinPlanner/Core/Models/ExpenseCategory.swift
import SwiftUI

/// Built-in spending categories. Stored on `ExpenseItem` by raw value so the
/// enum can evolve without a Core Data style migration.
enum ExpenseCategory: String, CaseIterable, Identifiable, Codable {
    case groceries
    case transport
    case housing
    case entertainment
    case health
    case shopping
    case education
    case subscriptions
    case other

    var id: String { rawValue }

    var title: String {
        switch self {
        case .groceries: return "Продукты"
        case .transport: return "Транспорт"
        case .housing: return "Жильё"
        case .entertainment: return "Развлечения"
        case .health: return "Здоровье"
        case .shopping: return "Покупки"
        case .education: return "Образование"
        case .subscriptions: return "Подписки"
        case .other: return "Другое"
        }
    }

    var symbolName: String {
        switch self {
        case .groceries: return "cart.fill"
        case .transport: return "car.fill"
        case .housing: return "house.fill"
        case .entertainment: return "popcorn.fill"
        case .health: return "heart.fill"
        case .shopping: return "bag.fill"
        case .education: return "book.fill"
        case .subscriptions: return "repeat.circle.fill"
        case .other: return "ellipsis.circle.fill"
        }
    }

    var tint: Color {
        switch self {
        case .groceries: return .categoryGroceries
        case .transport: return .categoryTransport
        case .housing: return .categoryHousing
        case .entertainment: return .categoryEntertainment
        case .health: return .categoryHealth
        case .shopping: return .categoryShopping
        case .education: return .categoryEducation
        case .subscriptions: return .categorySubscriptions
        case .other: return .categoryOther
        }
    }
}

File: src/core/expenseCategory.js
export const ExpenseCategory = Object.freeze({
  groceries: 'groceries',
  transport: 'transport',
  housing: 'housing',
  entertainment: 'entertainment',
  health: 'health',
  shopping: 'shopping',
  education: 'education',
  subscriptions: 'subscriptions',
  other: 'other',
});

export const ALL_CATEGORIES = Object.values(ExpenseCategory);

const META = {
  [ExpenseCategory.groceries]: { title: 'Продукты', icon: '🛒', tint: '#34C759' },
  [ExpenseCategory.transport]: { title: 'Транспорт', icon: '🚌', tint: '#0A84FF' },
  [ExpenseCategory.housing]: { title: 'Жильё', icon: '🏠', tint: '#FF9F0A' },
  [ExpenseCategory.entertainment]: { title: 'Развлечения', icon: '🎬', tint: '#BF5AF2' },
  [ExpenseCategory.health]: { title: 'Здоровье', icon: '💊', tint: '#FF375F' },
  [ExpenseCategory.shopping]: { title: 'Покупки', icon: '🛍️', tint: '#5AC8FA' },
  [ExpenseCategory.education]: { title: 'Образование', icon: '📚', tint: '#FFD60A' },
  [ExpenseCategory.subscriptions]: { title: 'Подписки', icon: '🔁', tint: '#64D2FF' },
  [ExpenseCategory.other]: { title: 'Другое', icon: '✨', tint: '#8E8E93' },
};

export function categoryTitle(category) {
  return META[category]?.title ?? META[ExpenseCategory.other].title;
}

export function categoryIcon(category) {
  return META[category]?.icon ?? META[ExpenseCategory.other].icon;
}

export function categoryTint(category) {
  return META[category]?.tint ?? META[ExpenseCategory.other].tint;
}
